packages/core/src/awsService/s3/commands/uploadFile.ts

```diff
@@ -437,7 +437,7 @@ export async function promptUserForBucket(
         }
     })
 
-    const lastTouchedFolder = globals.context.globalState.get<SavedFolder | undefined>('aws.lastTouchedS3Folder')
+    const lastTouchedFolder = globals.globalState.tryGet<SavedFolder>('aws.lastTouchedS3Folder', Object)
```

Comment:
`tryGet` checks that the stored value is actually an object, unlike vscode's `get()`. This avoids bugs like https://github.com/aws/aws-toolkit-vscode/pull/3060#discussion_r1050240283